server/asset/btc/utxo.go

```diff
@@ -287,7 +287,7 @@ func (utxo *UTXO) Confirmations(context.Context) (int64, error) {
 		// See if we can find the utxo in another block.
 		newUtxo, err := utxo.btc.utxo(&utxo.tx.hash, utxo.vout, utxo.redeemScript)
 		if err != nil {
-			return -1, fmt.Errorf("utxo block is not mainchain")
+			return -1, fmt.Errorf("utxo error: %w", err)
```

Comment:
`immature output` was one of the ones left after the bug fixes.  Nothing to do with mainchain anyway.  We might rethink the `utxo` method a little because there is a `maturity` field so I'm not sure it should error instead of deferring to the caller to decide if immature is a problem (server is probably checking funding coins or swap txns, so it does make sense, but in terms of general operation `utxo` failing to provide a return for immature outputs even though it has a maturity field is strange)